tests/Parsing/EntityCommentTest.elm

```diff
@@ -0,0 +1,243 @@
+
+        checkExpectation expected modString =
+            case TestUtil.stringToIntermediate modString of
+                Just m ->
+                    m.entities
+                        |> List.head
+                        |> Maybe.map
+                            (\ent -> Expect.equal ent.comment expected)
+                        |> Maybe.withDefault
+                            (Expect.fail "No entities parsed.")
+
+                Nothing ->
+                    Expect.fail "Failed to parse the module!"
+    in
+    Test.describe "Test the parsing function for the declaration comments."
+        [ Test.describe "Modules with no declarations."
+            (List.indexedMap
+                (\idx ->
+                    \moduleStr ->
+                        Test.test
+                            ("no declarations " ++ toString idx)
+                            (\() -> checkNoEntities moduleStr)
+                )
+                [ emptyModuleWithDoc
+                , emptyModuleWithoutDoc
+                , moduleNoDeclarationsWithDoc
+                , moduleNoDeclarationsWithoutDoc
+                ]
+            )
+        , Test.describe "Modules with declarations and no comment."
+            (List.indexedMap
+                (\idx ->
+                    \moduleStr ->
+                        Test.test
+                            ("declarations with no comments " ++ toString idx)
+                            (\() -> checkExpectation Nothing moduleStr)
+                )
+                [ moduleNoImportsWithDoc
+                , moduleNoImportsWithoutDoc
+                ]
+            )
+        , Test.describe
+            "Modules with wrong comment format, which does not get parsed."
+            (List.indexedMap
+                (\idx ->
+                    \moduleStr ->
+                        Test.test
+                            ("declarations with no comments " ++ toString idx)
+                            (\() -> checkExpectation Nothing moduleStr)
+                )
+                [ moduleWithWrongTypeDocFormat1
+                , moduleWithWrongTypeDocFormat2
+                ]
+            )
+        , Test.test "Module with documented type 1." <|
+            \() ->
+                checkExpectation (expectedComment 2 0 3 2)
+                    moduleWithDocumentedTypeAndNoDoc
+        , Test.test "Module with documented type 2." <|
+            \() ->
+                checkExpectation (expectedComment 5 0 6 2)
+                    moduleWithDocumentedTypeAndDoc
+        , Test.test "Module with documented type 3." <|
+            \() ->
+                checkExpectation (expectedComment 9 0 10 2)
+                    moduleWithDocumentedTypeAndImportsAndDoc
+        , Test.test "Module with documented type 4." <|
+            \() ->
+                checkExpectation (expectedComment 11 0 12 2)
+                    largerExample
+        ]
+
+
+emptyModuleWithDoc : String
```

Comment:
Please in-line these strings for readability of the tests. You might need to use a dedent function so that you can nicely indent the strings to conform with the indention of the code.